### ISSUE TITLE
Running this plugin stops all other loaders

### ISSUE THREAD

**Issue by ripperdoc** [cascornelissen/svg-spritemap-webpack-plugin#23]
I am running "svg-spritemap-webpack-plugin": "^2.0.0-rc.4" on webpack 4.1.1. My config for this plugin is:
```
        new SVGSpritemapPlugin({
            src: 'assets/gfx/*.svg',
            filename: 'spritemap.[contenthash].svg',
            prefix: 'fablr-',
        }),
```
But when this is active, all other files (JS, CSS, etc) are not emitted anymore. So it's unusable for me. I'm not able to see what happens, except that everything is back to normal when I comment out SVGSpritemapPlugin. Any pointers?

**Comment by cascornelissen** [comment]
Thanks for reporting the issue, this seems like a serious problem and I would love to get this fixed. Could you answer a couple of questions for me?

1. What exactly do you mean by "not emitted anymore"? Do these files not get created?
1. Does this only happen in a specific mode (watch, development, production)?
1. Do older RCs of the plugin work for you?
1. Have you used the `1.x` version of the plugin before upgrading to webpack 4? If so, did that work for you without any problems?
1. What OS/NodeJS/npm versions are you on?

It would be even better if you could provide a simple project where you can reproduce this, otherwise could you add your entire `webpack.config.js` to this issue?

Let's see what we can make of this 😄

**Comment by ripperdoc** [comment]
Let me try to give a bit more info, I've cut out almost everything else in my project and it still persists. 

Some details below, short answers to your questions:

1. Well, without SVGSpritemapPlugin, the expected JS files are emitted to the output directory. With the plugin, no JS files are emitted, just an SVG.
2. Seems to happen in both
3. As I check this, it only happens in rc.4, not 3.
4. I have not, sorry
5. OSX/Node 9.8.0/npm 5.6.0

Here is my config:
```
const SVGSpritemapPlugin = require('svg-spritemap-webpack-plugin');
const path = require('path');

config = {
    mode: 'production',
    entry: {
        app: path.resolve(__dirname, 'assets/js/test.js'),
    },
    output: {
        path: path.resolve(__dirname, 'static/dist'),
        filename: '[name].[chunkhash].js',
        publicPath: '/static/dist/'
    },
    plugins: [
        new SVGSpritemapPlugin({
            src: 'assets/gfx/*.svg',
            filename: 'spritemap.[contenthash].svg',
            prefix: 'fablr-',
        }),
    ],
    resolve: {
        modules: ['assets/js', 'node_modules'],
    }
}

module.exports = config
```
Here is my test.js entrypoint:
```
var utils = require('utils');
```
(Utils is just some general JS code to have something real)

The output from running with SVGSpritemapPlugin:
```
> webpack -p

Hash: 9cfb373ec8206974ead6
Version: webpack 4.1.1
Time: 331ms
Built at: 3/11/2018 10:28:48 AM
                         Asset      Size  Chunks             Chunk Names
spritemap.0ddc6c87790465ed.svg  18.9 KiB       0  [emitted]  spritemap
```
The output if I comment it out:
```
> webpack -p

Hash: c383ec3ee00f58db5f10
Version: webpack 4.1.1
Time: 1499ms
Built at: 3/11/2018 10:32:09 AM
 Asset      Size  Chunks             Chunk Names
app.js  19.3 KiB       0  [emitted]  app
Entrypoint app = app.js
   [1] ./assets/js/utils.js 6.33 KiB {0} [built]
   [2] ./assets/js/test.js 769 bytes {0} [built]
    + 1 hidden module
```

**Comment by cascornelissen** [comment]
I might have a problem where this is coming from, could you change your entry to this:

```
entry: {
    app: [path.resolve(__dirname, 'assets/js/test.js')],
},
```

The string-variant should work as well so I'll have to fix it but let's check if this is your actual problem first. Could you report back if this is working for you?

**Comment by ripperdoc** [comment]
That does not seem to make any difference.

**Comment by cascornelissen** [comment]
I'll have to look into this some more then, for now you could use `2.0.0-rc.3`.

**Comment by cascornelissen** [comment]
@ripperdoc, I'm having trouble reproducing your errors but could you try modifying the `node_modules/svg-spritemap-webpack-plugin/index.js` file in your project and changing [this line](https://github.com/cascornelissen/svg-spritemap-webpack-plugin/compare/v2.0.0-rc.3...v2.0.0-rc.4#diff-6d186b954a58d5bb740f73d84fe39073R120) to `return;` instead of `return true;` and see if that solves the problem for you?

**Comment by ripperdoc** [comment]
I made the edit, but the error seem to persist unfortunately.

**Comment by cascornelissen** [comment]
That's strange... The only difference between `rc.3` and `rc.4` are [changes to the svg4everybody helper](https://github.com/cascornelissen/svg-spritemap-webpack-plugin/compare/v2.0.0-rc.3...v2.0.0-rc.4) and since you're not using the helper (it defaults to `false`) the only difference for you would be the chance I [suggested earlier](https://github.com/cascornelissen/svg-spritemap-webpack-plugin/issues/23#issuecomment-372103239).

Any chance you could check that change again or dive into it with help of the changes between `rc.3` and `rc.4`? I've reproduced this about an hour ago and it was fixed with the change I suggested so it would be great if you could check again.

**Comment by cascornelissen** [comment]
@ripperdoc, I did some more testing and that I can't reproduce your issue when applying the change I've described earlier. I've pushed that change to GitHub, could you try installing the plugin directly from GitHub and report back if that fixes it?

```shell
npm install cascornelissen/svg-spritemap-webpack-plugin#master --save-dev
```

**Comment by vzaidman** [comment]
@ripperdoc

**Comment by cascornelissen** [comment]
Since I'm noticing multiple people responding to this issue (@Haydencleary, @vzaidman), is there anyone experiencing this issue as well?

If not, I'll release `2.0.0` this weekend either way.

**Comment by vzaidman** [comment]
testing

**Comment by cascornelissen** [comment]
@vzaidman, any updates by any chance? I'll release `2.0.0` tomorrow if I'm not hearing back from anyone.

**Comment by ripperdoc** [comment]
Sorry, I haven't had chance to test yet, but will get to it eventually.

**Comment by vzaidman** [comment]
ill test it in a few hours.

**Comment by ripperdoc** [comment]
I tested the master version and it seems to fix it! Thanks!

**Comment by vzaidman** [comment]
looks fine here as well. other plugins prevent me from migrating to webpack 4 right now, so i cant see if it actually works, but it looks like its fine right now. thanks :)

**Comment by cascornelissen** [comment]
@ripperdoc, great to hear! I've just released [`v2.0.0`](https://github.com/cascornelissen/svg-spritemap-webpack-plugin/releases/tag/v2.0.0) which includes this fix 🚀
